src/Native/Unix/System.IO.Ports.Native/pal_serial.h

```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_types.h"
+#include "pal_compiler.h"
+
+BEGIN_EXTERN_C
+
+intptr_t SystemIoPortsNative_SerialPortOpen(const char * name);
```

Comment:
Prefix misssing:
`DLLEXPORT intptr_t`

same for add `SystemIOPortNative_*` in `.h` files only.